include/gridtools/stencil_composition/backend_mc/loops.hpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include <type_traits>
+#include <utility>
+
+#include "../../common/defs.hpp"
+#include "../../common/generic_metafunctions/for_each.hpp"
+#include "../../common/tuple_util.hpp"
+#include "../../meta.hpp"
+#include "../dim.hpp"
+#include "../sid/concept.hpp"
+#include "execinfo_mc.hpp"
+
+namespace gridtools {
+    namespace mc {
+        namespace loops_impl_ {
+            template <class Stage, class Ptr, class Strides>
+            GT_FORCE_INLINE void i_loop(int_t size, Stage stage, Ptr &ptr, Strides const &strides) {
+#ifdef NDEBUG
+#pragma ivdep
+#ifndef __INTEL_COMPILER
```

Comment:
Why this #ifndef __INTEL_COMPILER? Did it generate code that did not validate? Maybe we have to re-run tests with different combinations of pragmas on different compilers, the current set of pragmas is at the border of legality for the present code, so maybe we can find a better option (probably after merge).

Reply:
I've added you comment a TODO